sql: refactor some SQL functions

madelynnblue opened the pull request:
This is in preparation for merging the some work currently on a branch, which already has
this change.

Newly exported names: sql.MakeTableDesc, sql.MustGetDatabaseDesc,
sql.ProcessDefaultColumns, sql.GenerateInsertRow, sql.RowInserter,
sql.MakeRowInserter. There are also a few new interfaces for dealing
with RowInserter in a more general way. These have all been added to
split out some of the create and insert logic so that it can be used in
other paths that aren't direct insertion into a SQL table, but instead
allow us to simulate the creation and insertion of tables.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/11758)
<!-- Reviewable:end -->

a-robinson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I sure wish reviewable could detect when code was moved from one place to another rather than treating it as a bunch of adds and deletes.

---

Reviewed 7 of 8 files at r1.
Review status: 7 of 8 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/create.go, line 971 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-KXwCczoGV0mtNl8G1kw:-KXwCczoGV0mtNl8G1kx:byxmxeq) ([raw file](https://github.com/cockroachdb/cockroach/blob/fba0d4c014bbe603ba8e4b1a6a98131352eab7a8/pkg/sql/create.go#L971)):*
> ```Go
> 	index *sqlbase.IndexDescriptor,
> 	interleave *parser.InterleaveDef,
> 	database string,
> ```

`database` doesn't quite capture the fact that this is the session's default database, not necessarily the database in which we'll be adding an interleave. It's probably worth renaming to something like `sessionDB`

---

*[pkg/sql/create.go, line 1187 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-KXwCtGodEL84nEkXpvo:-KXwCtGodEL84nEkXpvp:bhyn304) ([raw file](https://github.com/cockroachdb/cockroach/blob/fba0d4c014bbe603ba8e4b1a6a98131352eab7a8/pkg/sql/create.go#L1187)):*
> ```Go
> 	privileges *sqlbase.PrivilegeDescriptor,
> 	affected map[sqlbase.ID]*sqlbase.TableDescriptor,
> 	database string,
> ```

same naming comment here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-:-KXwE9uMDKWgiYSxgeU5:b-lr1psr)*
<!-- Sent from Reviewable.io -->

tamird commented:
It would be great if you could expand on what's being done and why in this commit message rather than just saying "we did it elsewhere".

---

Review status: 7 of 8 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-:-KXwLB2uAQ2-kqFQcE6s:bak459a)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: refactor some SQL functions

This is in preparation for merging the some work currently on a branch,
which already has this change.

Newly exported names: sql.MakeTableDesc, sql.MustGetDatabaseDesc,
sql.ProcessDefaultColumns, sql.GenerateInsertRow, sql.RowInserter,
sql.MakeRowInserter. There are also a few new interfaces for dealing
with RowInserter in a more general way. These have all been added to
split out some of the create and insert logic so that it can be used in
other paths that aren't direct insertion into a SQL table, but instead
allow us to simulate the creation and insertion of tables.

mjibson commented:
Added comments about what this will be used for.

---

Review status: 6 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/create.go, line 971 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-KXwCczoGV0mtNl8G1kw:-KY-gZMWyBwbVKy7ZjGF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fba0d4c014bbe603ba8e4b1a6a98131352eab7a8/pkg/sql/create.go#L971)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

`database` doesn't quite capture the fact that this is the session's default database, not necessarily the database in which we'll be adding an interleave. It's probably worth renaming to something like `sessionDB`
</blockquote></details>

Done.

---

*[pkg/sql/create.go, line 1187 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-KXwCtGodEL84nEkXpvo:-KY-gZwiHjigWl8c88ca:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fba0d4c014bbe603ba8e4b1a6a98131352eab7a8/pkg/sql/create.go#L1187)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

same naming comment here
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/11758#-:-KY-gbIXkDs-qS3TnXiK:bcxi0ux)*
<!-- Sent from Reviewable.io -->
